sky/skylet/providers/oci/node_provider.py

```diff
@@ -0,0 +1,490 @@
+"""OCI Node Provider.
+
+Node provider is called by the Ray Autoscaler to provision new compute
+resources (head / worker nodes).
+
+To show debug messages, export SKYPILOT_DEBUG=1 
+
+History:
+ - Hysun He (hysun.he@oracle.com) @ Apr, 2023: Initial implementation
+ 
+"""
+
+import logging
+import oci
```

Comment:
We should use the `oci_adaptor` instead of directly importing this. Otherwise, the user will get error when `oci` is not installed.